### PR TITLE
Add fish shell detection

### PR DESCRIPTION
**Pull request by Qinusty**
Implement basic shell detection for Linux platforms in the `kind create` command. 

### Additions:
- Implement `GetLinuxShell()` utility function
- Add fish as a custom output whilst leaving bash/zsh under default

### CODE REVIEW

File: pkg/internal/cluster/create/create.go [hunk 1]
```diff
@@ -190,12 +191,20 @@ func printUsage(name string) {
 			name,
 		)
 	} else {
+		// Detect shell and usage build prompt appropriately.
+		setKubeConfigPrompt := ""
+		switch shell := env.GetLinuxShell(); shell {
```
Comment (neolit123) [inline pkg/internal/cluster/create/create.go]: shouldn't the function be called `UnixShell`, as this branch is `!windows`?
can it cover custom OSX shells too?

**Comment by neolit123** [comment]
thanks for the PR @Qinusty 

please make sure you sign the CLA if this command fails:
/check-cla

**Comment by neolit123** [comment]
if you have signed the CLA already, make sure your CLA email is added to your github account.
if there is a problem with that try contacting the CNCF help-desk.

WRT this PR, i must admit this is the first time i've heard about "fish" shell, and given all the flavors of shells out there, it might be better to not add special handling for all of them?
/assign @BenTheElder

File: pkg/internal/cluster/create/create.go [hunk 2]
```diff
@@ -190,12 +191,20 @@ func printUsage(name string) {
...
+		case "fish":
+			setKubeConfigPrompt = "set -x KUBECONFIG (kind get kubeconfig-path --name=%q)"
```
Comment (BenTheElder) [inline pkg/internal/cluster/create/create.go]: I'm a bit dumbfounded that a Linux shell doesn't support POSIX... 🧐
https://pubs.opengroup.org/onlinepubs/009695399/utilities/export.html

File: pkg/internal/cluster/create/create.go (hunk 1 above)
Comment (BenTheElder) [reply pkg/internal/cluster/create/create.go]: also noting that fish appears to be on macOS

File: pkg/internal/util/env/shell.go [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package env
+
+import (
+	"os"
+	"strings"
+)
+
+func GetLinuxShell() string {
+	shell := os.Getenv("SHELL")
```
[the remark this answers is not in the document]
Comment (Qinusty) [reply pkg/internal/util/env/shell.go]: This is currently a best guess, I'm not certain of a guaranteed way to get the users current shell. Scraping current running processes would run into a similar issue if the user was running multiple shells.

File: pkg/internal/cluster/create/create.go (hunk 2 above)
Comment (Qinusty) [reply pkg/internal/cluster/create/create.go]: Yeah it's a bit strange but it appears to be the case. https://fishshell.com/docs/current/tutorial.html#tut_exports

**Comment by BenTheElder** [comment]
#850 landed so we don't need to export this env anymore.

Thanks for the PR though!

**Pull request closed without merging**